packages/template-retail-react-app/app/utils/url.test.js

```diff
@@ -419,3 +420,50 @@ describe('removeSiteLocaleFromPath', function () {
         expect(pathName).toBe('')
     })
 })
+
+describe('serverSafeEncode', () => {
```

Comment:
Only other thing I can think of is the gross double-encoding scenario, so that would be a test case like

```
    test('correctly double encodes', () => {
        const input = 'My%20Address%20%26%20Co.'
        const encoded = serverSafeEncode(input)

        // Decode should give us original string
        const decoded = decodeURIComponent(encoded)
        expect(decoded).toBe(input)
    })
```